lib/chef/knife/vsphere_datastore_list.rb

```diff
@@ -34,8 +34,10 @@ class Chef::Knife::VsphereDatastoreList < Chef::Knife::BaseVsphereCommand
          description: 'Target pool'
 
   def find_pools(folder, poolname = nil)
-    pools = folder.children.find_all.select { |p| p.is_a?(RbVmomi::VIM::ComputeResource) || p.is_a?(RbVmomi::VIM::ResourcePool) }
-    poolname.nil? ? pools : pools.select { |p| p.name == poolname }
+    pools = traverse_folders_for_pools(folder)
```

Comment:
turns out the current implementation does not display clusters/pools that are nested inside of folders. leveraging the traverse functions from the base class allows them to be included. this issue also exists in `hosts_list`